olive/cli/quantize.py

```diff
@@ -0,0 +1,133 @@
+# --------------------------------------------------------------------------
+
+# ruff: noqa: T201
+# ruff: noqa: RUF012
+
+import tempfile
+from argparse import ArgumentParser
+from copy import deepcopy
+from typing import Any, Dict
+
+from olive.cli.base import (
+    BaseOliveCLICommand,
+    add_accelerator_options,
+    add_dataset_options,
+    add_input_model_options,
+    add_logging_options,
+    add_remote_options,
+    is_remote_run,
+    save_output_model,
+    update_accelerator_options,
+    update_dataset_options,
+    update_input_model_options,
+)
+from olive.common.utils import set_nested_dict_value
+
+
+class QuantizeCommand(BaseOliveCLICommand):
+    _CONFIG_TEMPLATE = {
+        "input_model": {"type": "HfModel", "load_kwargs": {"attn_implementation": "eager"}},
+        "systems": {
+            "local_system": {
+                "type": "LocalSystem",
+                "accelerators": [{"device": "gpu", "execution_providers": ["CUDAExecutionProvider"]}],
+            }
+        },
+        "data_configs": [
+            {
+                "name": "default_data_config",
+                "type": "HuggingfaceContainer",
+                "load_dataset_config": {},
+                "pre_process_data_config": {},
+                "dataloader_config": {},
+                "post_process_data_config": {},
+            }
+        ],
+        "passes": {
+            "awq": {"type": "AutoAWQQuantizer"},
+            "gptq": {
+                # Ref: https://github.com/AutoGPTQ/AutoGPTQ/pull/651/files
+                "type": "GptqQuantizer",
+                "data_config": "default_data_config",
+            },
+            "quarot": {
+                "type": "QuaRot",
+                "w_rtn": True,
+                "rotate": True,
+                "w_bits": 4,
+                "a_bits": 4,
+                "k_bits": 4,
+                "v_bits": 4,
+                "calibration_data_config": None,
+            },
+        },
+        "pass_flows": [],
+        "output_dir": "models",
+        "host": "local_system",
+        "target": "local_system",
+    }
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "quantize",
+            help="Quantize the input model",
+        )
+
+        # model options
+        add_input_model_options(sub_parser, enable_hf=True, enable_pt=True, default_output_path="quantized-model")
+
+        # Logging options
+        add_logging_options(sub_parser)
+
+        sub_parser.add_argument(
+            "--algorithms",
+            type=str,
+            nargs="*",
+            required=True,
+            choices=sorted(QuantizeCommand._CONFIG_TEMPLATE["passes"].keys()),
+            help="List of quantization algorithms to run.",
+        )
+
+        # dataset options
+        add_dataset_options(sub_parser, include_train=False, include_eval=False)
+
+        # accelerator options
+        add_accelerator_options(sub_parser)
```

Comment:
since only pytorch models are currently supported, I don't think this is needed? The ep+device doesn't matter for pytorch only passes. they are just placeholders.

Reply:
even device is not necessary. if the pass needs to run on gpu, it will run on gpu. 

might not be an issue for cpu device but if you choose gpu, with ep not provided, the behavior is that it will loop over all gpu related eps. not necessary here

Reply:
oh, the looping doesn't happen since the template has one ep already in the system. but regardless, I think the accelerator option is not needed for now. it is confusing when they don't do anything.

Reply:
Wouldn't that equate to removing _systems_, _host_ and _target_ settings from config?? If I include those options then I have to choose either cpu or gpu explicitly, which might be right for one algorithm but not the other.

Reply:
since the only tools that it currently supports are pytorch quantization tools, I think it's okay to just keep the current default system as is in the config. they are just there as placeholders. 
- the target accelerator spec doesn't matter for pytorch passes but we need some placeholder to ensure the engine doesn't do an unnecessary loop of all possible accelerators.
- accelerator spec for host device also doesn't really make sense. If it needs to run on gpu, it should run on gpu. I think Mike added it, but its usage is arbitrary so far.
Most of these are artifacts of our originally ort focused design decisions.

Reply:
you can keep the options if you want. just think it's unnecessary and it currently works fine because the config already has a default ep that prevents the looping behavior.